docs/native-modules-setup.md

```diff
@@ -45,154 +45,64 @@ The steps to create a new native module library project are:
-| JSI.Universal                       | `JSI\Universal\JSI.Universal.vcxproj`                                            |
-| Microsoft.ReactNative               | `Microsoft.ReactNative\Microsoft.ReactNative.vcxproj`                            |
-| Microsoft.ReactNative.Cxx           | `Microsoft.ReactNative.Cxx\Microsoft.ReactNative.Cxx.vcxitems`                   |
-| Microsoft.ReactNative.SharedManaged | `Microsoft.ReactNative.SharedManaged\Microsoft.ReactNative.SharedManaged.shproj` |
-| Mso                                 | `Mso\Mso.vcxitems`                                                               |
-| ReactCommon                         | `ReactCommon\ReactCommon.vcxproj`                                                |
-| ReactWindowsCore                    | `ReactWindowsCore\ReactWindowsCore.vcxproj`                                      |
-| Shared                              | `Shared\Shared.vcxitems`                                                         |
-
-For each project, you'll do the following:
+```bat
+npm info react-native-windows@canary devDependencies.react
+```
 
-1. Open the Solution Explorer sidebar.
-1. Right-click on the `ReactNative` folder.
-1. Select `Add` > `Existing Project...`.
-1. Select the project file and click `Open`.
+Take the result of that command (let's say it's `16.13.1`) and use it to upgrade the dev dependency:
 
-When you are done, your solution should look like this:
-![native module dependencies](assets/native-module-dependencies.png)
+```bat
+yarn upgrade react@16.13.1 --dev
+```
 
-You now have all of the React Native Windows projects to your solution. Next we're going to reference them in our `MyLibrary` project.
+You'll need to repeat the steps for `react-native`, ie:
 
-### Referencing React Native Windows in your Project
+```bat
+npm info react-native-windows@canary devDependencies.react-native
+```
 
-The only project reference you **must** add is `Microsoft.ReactNative`. To add the reference:
+Again, take the result of that command (let's say it's `0.0.0-a36d9cd7e`) and use it to upgrade the dev dependency:
 
-1. Open the Solution Explorer sidebar.
-1. Right-click on your `MyLibrary` project.
-1. Select `Add` > `Reference`.
-1. Select `Projects` on the left-hand side.
-1. Check the box next to `Microsoft.ReactNative`.
-1. Click `OK`.
+```bat
+yarn upgrade react-native@0.0.0-a36d9cd7e --dev
+```
 
-After you've added the reference, you need to make sure it doesn't copy itself into your build (otherwise it'll cause build conflicts down the line when you're trying on consume your library):
+Now you should be ready to add Windows support with the CLI.
 
-1. Open the Solution Explorer sidebar.
-1. Under your `MyLibrary` project, expand the `References`.
-1. Right-click on `Microsoft.ReactNative`.
-1. Select `Properties`.
-1. Under `Build`, Change `Copy Local` to `False`.
+## Adding Windows support to an existing library
```

Comment:
Does `create-react-native-module` add example projects for iOS/Android that would need to be revved?

Reply:
`create-react-native-module` *can* create example projects with a flag.

I'm not sure what guidance to give about the main project's dependencies, let alone example app dependencies. The instructions here make sure that you have the right files on disk to build. And even if you publish to npm (with or without the changes to package.json), an RNW app will be able to consume your native module.

This is least intrusive way to implement modules without any assumptions or proscriptions about the module's maintenance.

Reply:
My concern was more that someone could read this, then rev existing examples versions without necessarily keeping them up to date. I.e. if we create iOS/Android examples for a version, then run these commands, there is some extra work to do still that has tripped people up before.

Could we add a sentence pointing folks towards upgrade-helper if there are a existing sample applications?